test/addons-abi/test_buffer/test_buffer.cc

```diff
@@ -0,0 +1,141 @@
+#include <string.h>
+#include <string>
+#include <node_api_helpers.h>
+
+#define JS_ASSERT(env, assertion, message) \
+  if (!(assertion)) { \
+    napi_throw_error((env), (std::string("assertion (" #assertion ") failed: ") + message).c_str()); \
+	return; \
+  }
+
+#define NAPI_CALL(env, theCall) \
+  if (theCall != napi_ok) { \
+    const char *errorMessage = napi_get_last_error_info()->error_message; \
+	errorMessage = errorMessage ? errorMessage: "empty error message"; \
+    napi_throw_error((env), errorMessage); \
+	return; \
+  }
+
+static const char theText[] =
+  "Lorem ipsum dolor sit amet, consectetur adipiscing elit.";
+
+static int deleterCallCount = 0;
+static void deleteTheText(void *data) {
+  delete (char *)data;
+  deleterCallCount++;
+}
+
+static void noopDeleter(void *data) {
+  deleterCallCount++;
+}
+
+NAPI_METHOD(newBuffer) {
+  napi_value theBuffer;
+  char *theCopy = strdup(theText);
+  JS_ASSERT(env, theCopy,
+    "Failed to copy static text for newBuffer");
+  NAPI_CALL(env, napi_create_buffer(env, theCopy, sizeof(theText),
+    deleteTheText, &theBuffer));
+  NAPI_CALL(env, napi_set_return_value(env, info, theBuffer));
+}
+
+NAPI_METHOD(getDeleterCallCount) {
+  napi_value callCount;
+  NAPI_CALL(env, napi_create_number(env, deleterCallCount, &callCount));
+  NAPI_CALL(env, napi_set_return_value(env, info, callCount));
+}
+
+NAPI_METHOD(copyBuffer) {
+  napi_value theBuffer;
+  NAPI_CALL(env, napi_create_buffer_copy(env, theText, sizeof(theText),
+    &theBuffer));
+  NAPI_CALL(env, napi_set_return_value(env, info, theBuffer));
+}
+
+NAPI_METHOD(bufferHasInstance) {
+  int argc;
+  NAPI_CALL(env, napi_get_cb_args_length(env, info, &argc));
+  JS_ASSERT(env, argc == 1, "Wrong number of arguments");
+  napi_value theBuffer;
+  NAPI_CALL(env, napi_get_cb_args(env, info, &theBuffer, 1));
+  bool hasInstance;
+  NAPI_CALL(env, napi_is_buffer(env, theBuffer, &hasInstance));
```

Comment:
Can you also add a call to `napi_get_type_of_value()` on the buffer? The expected result should be `napi_object`, except I think it might fail with JSRT. Due to use of the shim there the buffer `napi_value` is not a `JsValueRef`, so `JsGetValueType()` will fail, or maybe crash.